core/utils/utils.go

```diff
@@ -531,6 +531,20 @@ func ContextFromChan(chStop <-chan struct{}) (context.Context, context.CancelFun
 	return ctx, cancel
 }
 
+// ContextFromChanWithDeadline creates a context with a deadline that finishes when the provided channel
+// receives or is closed.
+func ContextFromChanWithDeadline(chStop <-chan struct{}) (context.Context, context.CancelFunc) {
```

Comment:
this func name makes me think it would accept deadline as an arg. Can we do that instead of defaulting to 1 min?

Reply:
Done